pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTenantRestletResource.java

```diff
@@ -196,6 +200,52 @@ public String listInstanceOrToggleTenantState(
     }
   }
 
+  /**
+   * This method expects a tenant name and will return a list of tables tagged on that tenant. It assumes that the
+   * tagname is for server tenants only. This method will only read CURRENTSTATES in zk.
+   * @param tenantName
+   * @return
+   */
+  @GET
+  @Path("/tenants/{tenantName}/tables")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "List tables on a a server tenant")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"),
+      @ApiResponse(code = 500, message = "Error reading list")
+  })
+  public String getTablesOnTenant(
+      @ApiParam(value = "Tenant name", required = true) @PathParam("tenantName") String tenantName) {
+    return getTablesServedFromTenant(tenantName);
+  }
+
+  private String getTablesServedFromTenant(String tenantName)
+      throws JSONException {
+    Set<String> tables = new HashSet<String>();
+    JSONObject resourceGetRet = new JSONObject();
+    Set<String> instances = pinotHelixResourceManager.getAllInstancesForServerTenant(tenantName);
+
+    if (instances.size() == 0) {
```

Comment:
Do we enforce and ensure that all created tenants have instances? If not, then is this really an error situation?